aws/cf_template.go

```diff
@@ -268,10 +274,17 @@ func generateTemplate(spec *stackSpec) (string, error) {
 	})
 
 	if spec.loadbalancerType == LoadBalancerTypeApplication && spec.wafWebAclId != "" {
-		template.AddResource("WAFAssociation", &cloudformation.WAFRegionalWebACLAssociation{
-			ResourceArn: cloudformation.Ref("LB").String(),
-			WebACLID:    cloudformation.String(spec.wafWebAclId),
-		})
+		if strings.HasPrefix(spec.wafWebAclId, "arn:aws:wafv2") {
+			template.AddResource("WAF2WAC1KPI3", &cloudformation.WAFv2WebACLAssociation{
+				ResourceArn: cloudformation.Ref("LB").String(),
+				WebACLArn:   cloudformation.String(spec.wafWebAclId),
+			})
+		} else {
```

Comment:
What if we have no waf ID specified?
We should omit creating one as far as I understand.
Related https://github.com/zalando-incubator/kube-ingress-aws-controller/pull/341

Reply:
It won't get into the first if

> if spec.loadbalancerType == LoadBalancerTypeApplication && spec.wafWebAclId != "" {

* no waf, out of the entire if
* waf arn v2, template for v2
* waf id (v1), template for v1 as it used to be

---
But depending if #341 gets merged first this one has to be changed

waf2 (I suppose this)
> WebACLArn:    cloudformation.Ref(parameterLoadBalancerWAFWebACLIdParameter).String(),

waf1
> WebACLID:    cloudformation.Ref(parameterLoadBalancerWAFWebACLIdParameter).String(),

Reply:
I think this merge went wrong, see https://github.com/zalando-incubator/kube-ingress-aws-controller/issues/331#issuecomment-617031946

@janario @aryszka @mikkeloscar @szuecs